src/github/repository.ts

```diff
@@ -3,18 +3,25 @@ import { WebhookContext } from "routes/github/webhook/webhook-context";
 import { transformRepositoryId } from "~/src/transforms/transform-repository-id";
 import { Subscription } from "models/subscription";
 import { RepoSyncState } from "models/reposyncstate";
+import { findOrStartSync } from "~/src/sync/sync-utils";
 import { booleanFlag, BooleanFlags } from "config/feature-flags";
 
 
 export const repositoryWebhookHandler = async (context: WebhookContext, jiraClient, _util, gitHubInstallationId: number, subscription: Subscription): Promise<void> => {
 	if (context.action === "deleted") {
-		return deleteRepositoryWebhookHandler(context, jiraClient, gitHubInstallationId);
+		return deleteRepositoryWebhookHandler(context, jiraClient, gitHubInstallationId, subscription);
 	}
 	if (context.action === "created" && await booleanFlag(BooleanFlags.REPO_CREATED_EVENT, jiraClient.baseURL)) {
 		return createRepositoryWebhookHandler(context, gitHubInstallationId, subscription);
 	}
 };
 
+const updateRepoCount = async (subscription: Subscription) => {
+	// Update the repos by taking the new post-delete count of repos for the given subscription
+	const totalNumberOfRepos = (await RepoSyncState.findAllFromSubscription(subscription)).length;
```

Comment:
Actually seems there's a `RepoSyncState.countFromSubscription` ? Maybe we could use that?